src/panels/lovelace/elements/hui-state-icon-element.js

```diff
@@ -0,0 +1,51 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../../components/entity/state-badge.js';
+
+import ElementClickMixin from '../mixins/element-click-mixin.js';
+
+/*
+ * @appliesMixin ElementClickMixin
+ */
+class HuiStateIconElement extends ElementClickMixin(PolymerElement) {
+  static get template() {
+    return html`
+      <state-badge state-obj="[[_stateObj]]"></state-badge> 
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      _config: Object,
+      _stateObj: Object
+    };
+  }
+
+  ready() {
+    super.ready();
+    this.classList.add('clickable');
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('click', () => this.handleClick(this.hass, this._config));
```

Comment:
Doing this on `connected` is dangerous as a component could be connected multiple times.